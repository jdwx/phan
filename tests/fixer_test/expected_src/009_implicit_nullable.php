<?php
/**
 * @param ?array $y
 * @param ?int $z
 */
function test(?int $x = null, ?array $y = null, ?int $z = null) : void {
}
