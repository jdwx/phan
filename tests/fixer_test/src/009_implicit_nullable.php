<?php
/**
 * @param ?array $y
 * @param ?int $z
 */
function test(int $x = null, array $y = null, $z = null) {
}
