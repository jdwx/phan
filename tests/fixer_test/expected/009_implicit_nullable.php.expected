src_copy/009_implicit_nullable.php:6 PhanDeprecatedImplicitNullableParam Implicit nullable parameters (array $y = null) have been deprecated in PHP 8.4
src_copy/009_implicit_nullable.php:6 PhanDeprecatedImplicitNullableParam Implicit nullable parameters (int $x = null) have been deprecated in PHP 8.4
src_copy/009_implicit_nullable.php:6 PhanPluginCanUseNullableParamType Can use ?int as the type of parameter $z of test
src_copy/009_implicit_nullable.php:6 PhanPluginCanUsePHP71Void Can use php 7.1's void as a return type of test
src_copy/009_implicit_nullable.php:6 PhanUnreferencedFunction Possibly zero references to function \test()
src_copy/009_implicit_nullable.php:6 PhanUnusedGlobalFunctionParameter Parameter $x is never used
src_copy/009_implicit_nullable.php:6 PhanUnusedGlobalFunctionParameter Parameter $y is never used
src_copy/009_implicit_nullable.php:6 PhanUnusedGlobalFunctionParameter Parameter $z is never used
